controllers/flinkcluster_updater.go

```diff
@@ -205,6 +219,72 @@ func (updater *_ClusterStatusUpdater) deriveClusterStatus() flinkoperatorv1alpha
 			}
 	}
 
+	// (Optional) JobManager ingress.
+	var observedJmIngress = updater.observedState.jmIngress
+	if observedJmIngress != nil {
+		var state string
+		var urls []string
+		var useTLS bool
+
+		if len(observedJmIngress.Spec.TLS) > 0 {
+			useTLS = true
+		}
+
+		if useTLS {
+			for _, tls := range observedJmIngress.Spec.TLS {
+				for _, host := range tls.Hosts {
+					if host != "" {
+						urls = append(urls, "https://"+host)
```

Comment:
IIUC, urls represents the actual status, why do we include the host from the spec here? because below we use `len(urls) > 0` to determine the state of the Ingress, this makes it always Ready when host is provided, unable to detect failure. I think we should only include URLs from observedJmIngress.Status.LoadBalancer.Ingress.

Reply:
Thank you for your good feedback.

Firstly, the hostname of the ingress status is the hostname of the load balancer itself, so it will always provide the hostname of the LB separately from the host of the spec. Since the user provides the host in the ingress spec to access the host defined differently for each ingress resource, not the hostname of the LB, the URL intended by the user should be the host of the spec.

Secondly, as you say, the ready state is right by judging the status of an ingress resource. This is because the LB IP or hostname in ingress status is updated from blank to the actual LB value when the LB providing the ingress is determined. (And either IP or hostname is being used. Link: https://github.com/kubernetes/ingress-nginx/blob/04e2ad8fcd51b0741263a37b8e7424ca3979137c/internal/ingress/status/status.go#L218)

I initially tried to check the status as the ingress resource creation is complete because the ingress resource status may not be updated and may remain blank depending on the environment or method of installing the ingress controller. However, this is rare in a cloud environment, so in your opinion it seems reasonable to change to ready state when the LB value of status is filled. In order to properly check the status of ingress, it seems correct to configure the cluster so that the ingress status is updated on the cluster administrator side.

So what if we change two things?
1. The jobmanager ingress status is determined to be ready when the LB value of the ingress resource status is updated.
2. Specify the LB IP or hostname of ingress resource status in jobmanager ingress status urls only when jobmanager ingress hostFormat is not provided.

Note) There are some cases where the status of an ingress resource remains blank.
1.When installing with the commonly used helm chart stable / nginx-ingress and installing with default value without providing controller.publishService.enabled option (https://github.com/helm/charts/blob/master/ stable / nginx-ingress / values.yaml # L79)
2.If your ingress controller is configured to use the host network (https://github.com/kubernetes/ingress-nginx/blob/master/docs/deploy/baremetal.md)

Reply:
Thanks for your detailed explanation. The solution sounds good to me. I just tested it in Minikube, it worked. I'll test it in GKE tomorrow.